pyproject.toml

```diff
@@ -0,0 +1,36 @@
+[build-system]
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "xitorch"
+version = "0.4.0.dev20230208"
```

Comment:
On it.

Reply:
With the use of https://github.com/mtkennerly/poetry-dynamic-versioning this issue is resolved. New versions are now set by using tags, and development versions will automatically reference these tags.

Reply:
Sorry if I wasn't clear on the exact meaning. On Build, the version will check if a tag is set for the version. If it is, it will use the version of the tag, e.g.:

tag: `v0.4.0`
version: `0.4.0`

Otherwise, it will reference the next version of interest, and add a dev mark with a timestamp to it. Say the last version was `0.3.0`, it will tag the build as:
version: `0.4.0.devYYYYMMDDHHMMSS`
Alternatively, I can also easily implement this: `0.4.0.dev{commit_hash_abbrv}`